### PR TITLE
Candidate for release 3.2.0

### PR DESCRIPTION
**Pull request by kushti**
This PR contains candidate for release 3.2.0. It includes:

Main changes:

* PR #969 , #1011 - LevelDB is used now for storing the state (UTXO set + authenticated AVL+ tree on top of it)
* PR #1004 - a fix for light mode with blocksToKeep set (pruning switched on)

Additionally (fixes for minor issues not related to client functionality):
* PR #1010 - support for native-image generation in ergo-wallet
* PR #997  - from now every new release will be checked for bootstrapping in 3 full-node regimes (w. UTXO set, w/out UTXO set, w/out UTXO set and blockchain pruning)
* PR #1014 - improvements and fixes in ergo-installer.sh
* PR #1015 - fix for Docker build
* PR #1016 - fix for empty gitHeadCommit (minor building issue)

### CODE REVIEW

**Comment by greenhat** [comment]
@kushti I added `release/*` wildcard to it2 tests jenkins job.

File: src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala [hunk 1]
```diff
@@ -184,10 +186,13 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
...
   def appendHeader(header: Header, votingSettings: VotingSettings): Try[ErgoStateContext] = {
+    if (lastHeaders.isEmpty) {
+      log.info("Last headers are empty!")
+    }
     validateVotes(header)
       .validate(hdrHeight,
-        lastHeaderOpt.map(_.height + 1).getOrElse(ErgoHistory.GenesisHeight) == header.height,
-        s"${header.id} => ${lastHeaderOpt.map(_.height)} == ${header.height}")
+        lastHeaders.isEmpty || lastHeaders.head.height + 1 == header.height,
+        s"${header.id} => ${lastHeaderOpt.map(_.height + 1)} == ${header.height}")
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala]: these changes are suspicious, please provide explanation for the condition in the code.

File: src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala [hunk 2]
```diff
@@ -207,9 +212,8 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
     val votesValidationState = validateVotes(fb.header)
     new ExtensionValidator(votesValidationState)
       .validateExtension(fb.extension, fb.header, lastExtensionOpt, lastHeaderOpt)
-      .validate(hdrHeight,
-        lastHeaderOpt.map(_.height + 1).getOrElse(ErgoHistory.GenesisHeight) == fb.header.height,
-        s"${fb.id} => ${lastHeaderOpt.map(_.height)} == ${fb.header.height}")
+      .validate(hdrHeight,lastHeaders.isEmpty || lastHeaders.head.height + 1 == fb.header.height,
+        s"${fb.id} => ${lastHeaderOpt.map(_.height + 1)} == ${fb.header.height}")
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala]: Also description of logic would be helpful. also Why genesis check is no longer necessary.

**Comment by kushti** [comment]
@aslesarenko hdrHeight tabular check removed, as condition is different in history, 

made new method to check, its description:
"Checks that header is coming in order.
 If no last headers collected yet, header is always applicable. It could be genesis header if no pruning, or header at non-genesis height if pruning."

also, added some tests for new versioned storage

**Pull request merged**